test/app-tests/loading.js

```diff
@@ -616,11 +626,21 @@ describe('loading:', function() {
 
     // check that we have a Login component, send a 'user:pass' login,
     // and await the HTTP requests.
-    function completeLogin(matrixChat) {
+    async function completeLogin(matrixChat) {
         // we expect a single <Login> component
         const login = ReactTestUtils.findRenderedComponentWithType(
             matrixChat, sdk.getComponent('structures.auth.Login'));
 
+        // When we switch to the login component, it'll hit the login endpoint
+        // for proof of life and to get flows. We'll only give it one option.
+        httpBackend.when('GET', '/login')
+            .respond(200, {"flows": [{"type": "m.login.password"}]});
+        httpBackend.flush(); // We already would have tried the GET /login request
+
+        // Give the component some time to finish processing the login flows before
+        // continuing.
+        await Promise.delay(100);
```

Comment:
Hmm, usually adding a timeout like this is setting the test up to fail if it runs on very slow hardware, etc. Is there anything you could do instead, like perhaps wait for some event or element to appear? (It looks like we don't have a good answer to this problem for app tests today, so mostly just trying to get more people thinking about.)

Reply:
I'd love to wait for a component, but we don't have a reliable component to watch for changes.

Reply:
Hmm, so there's no visible change before and after processing the login flow?

Reply:
There's a loading icon, but it means far too many things. I suppose we could stick a class on it, but ultimately I'm tempted to just ignore the problem. We only really need 10/100 of those milliseconds, and slow machines are likely to have other problems than a random test.

Reply:
Well, in an ideal world, test wouldn't have any time dependence like this, but no easy solution for now it seems.